mllib/src/main/scala/org/apache/spark/mllib/classification/NaiveBayes.scala

```diff
@@ -83,9 +86,13 @@ class NaiveBayesModel private[spark] (
   }
 
   override def predict(testData: RDD[Vector]): RDD[Double] = {
-    val bcModel = testData.context.broadcast(this)
+    bcModel match {
```

Comment:
So you seem to have this code pattern repeated in a few places through the file, maybe we could factor this out into a common function.
Also this pattern is replicated across a few of your PRs, maybe adding a trait with this shared logic would be even more awesome?